File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        kv = {}
        
        for i, num in enumerate(nums):
            if target - num in kv.keys():
                return [i, kv[target-num]]
            else:
                kv[num] = i
                
        return []

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    numsMap := make(map[int]int)

    for i, num := range nums {
        complement := target - num

        // if complement in map
        if index, found := numsMap[complement]; found {
            return []int{index, i}
        }

        numsMap[num] = i

    }

    return nil
    
}